### PR TITLE
Add auto tests for `host.evacuate` method

### PR DESCRIPTION
**Pull request by benjamreis**
Auto test for this PR: https://github.com/xapi-project/xen-api/pull/4350.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -264,6 +270,12 @@ def additional_repos(request, hosts):
         for host_ in host.pool.hosts:
             host_.ssh(['rm', '-f', repo_file])
 
+@pytest.fixture(scope='session')
+def second_network(request):
+    network_uuid = request.param
+    assert network_uuid is not None, 'no second network provided'
```
Comment (stormi) [inline conftest.py:276]: Can this happen, given the test at line 309?

File: lib/common.py [hunk 2]
```diff
@@ -492,9 +492,13 @@ def is_suspended(self):
     def is_paused(self):
         return self.power_state() == 'paused'
 
-    def start(self):
+    # By xe design on must be an host name-label
+    def start(self, on=None):
         print("Start VM")
-        return self.host.xe('vm-start', {'uuid': self.uuid})
+        args = { 'uuid': self.uuid }
```
Comment (stormi) [inline lib/common.py:498]: I think the python codestyle we are more or less following doesn't add spaces between brackets and contents.

Comment (benjamreis) [reply lib/common.py:498]: Is there a python codestyle that we are more or less following? XD

File: tests/migration/test_host_evacuate.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import pytest
+
+from subprocess import CalledProcessError
+from lib.common import wait_for, wait_for_not
+# The pool needs a shared SR to use `host.evacuate`
+from tests.storage.nfs.conftest import *
```
Comment (stormi) [inline tests/migration/test_host_evacuate.py:6]: Wildcard import is discouraged by PEP 008 https://www.python.org/dev/peps/pep-0008/#imports

File: lib/common.py (hunk 2 above)
Comment (stormi) [reply lib/common.py:498]: Found it: https://www.python.org/dev/peps/pep-0008/#id26

There are several places where this comment applies.

File: tests/migration/test_host_evacuate.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+# Requirements:
+# From --hosts parameter:
+# - host(A1): first XCP-ng host >= 8.2 with an additional unused disk for the SR.
+# - hostA2: Second member of the pool. Can have any local SR. No need to specify it on CLI.
+# - second_network: A 2nd network of the pool, NOT the management interface, with PIF plugged on all hosts
```
Comment (stormi) [inline tests/migration/test_host_evacuate.py:12]: Isn't this is from  `--second-network` parameter?

File: tests/migration/test_host_evacuate.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        args['network-uuid'] = network_uuid
+
+    try:
+        source_host.xe('host-evacuate', args)
+        wait_for(lambda: vm.all_vdis_on_host(dest_host), "Wait for all VDIs on destination host")
+        wait_for(lambda: vm.is_running_on_host(dest_host), "Wait for VM to be running on destination host")
+        vm.wait_for_os_booted()
+        assert not expect_error, "host-evacuate should have raised: %s" % error
+    except CalledProcessError as e:
+        if not (expect_error and e.output.find(error) > -1) :
```
Comment (stormi) [inline tests/migration/test_host_evacuate.py:33]: extraneous space before `:`

File: tests/migration/test_host_evacuate.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        vm.shutdown(verify=True)
+
+def _save_ip_configuration_mode(host, pif_uuid):
+    args = { 'uuid': pif_uuid }
+    ipv6 = (host.xe('pif-param-get', {'uuid': pif_uuid, 'param-name': 'primary-address-type' }) == "IPv6")
+    keys = [
+        ('IPv6', 'IPv6'), ('DNS', 'DNS'), ('gateway', 'IPv6-gateway'), ('mode', 'IPv6-configuration-mode')
+    ] if ipv6 else [
+        ('IP', 'IP'), ('DNS', 'DNS'), ('gateway', 'gateway'), ('netmask', 'netmask'), ('mode', 'IP-configuration-mode')
+    ]
```
Comment (stormi) [inline tests/migration/test_host_evacuate.py:45]: Given that it's multiline, a simple `if` statement would seem cleared to me

File: tests/migration/test_host_evacuate.py (hunk 4 above)
Comment (benjamreis) [reply tests/migration/test_host_evacuate.py:12]: Yes why?

Comment (stormi) [reply tests/migration/test_host_evacuate.py:12]: It's in the "From --hosts parameter" section of the comment but it does not come from the `--hosts` parameter.

**Comment by stormi** [comment]
Looks good. Please squash commits 3&4 into commit 1 then we can merge.

**Comment by benjamreis** [comment]
Done

**Pull request merged**